src/modules/Login/verification/LoginverificationInput.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+
+import { useVerifyPhoneVerificationCodeMutation } from "src/graphql";
+import { useDebouncedCallback } from "src/hooks/useDebouncedCallback";
+import { TimeState } from "src/hooks/useTimer";
+import { Box } from "src/ui/Box";
+import { FormField } from "src/ui/Form";
+import { Input } from "src/ui/Input";
+import { Typography } from "src/ui/Typography";
+
+const WAITTING = 160000;
+const REGEXP_NUMBER = /[^0-9]/;
+
+interface LoginverificationProps {
+  setButtonDisabled: Dispatch<SetStateAction<boolean>>;
+  time: number;
+  timeState: TimeState;
+}
+
+type VerificationStateObject = {
+  [key in "initial" | "invalid" | "valid"]: {
+    color: "black5" | "danger1";
+    inputState?: "valid" | "invalid";
+    text?: string;
+  };
+};
+
+const verificationStateObject = (
+  timeState: TimeState,
+): VerificationStateObject => ({
+  initial: {
+    color: "black5",
+    inputState: undefined,
+    text: "인증번호가 전송되었습니다.",
+  },
+  invalid: {
+    color: "danger1",
+    inputState: "invalid",
+    text:
+      timeState === "end"
+        ? "인증시간을 초과하였습다. 재인증을 시도해주세요."
+        : "인증 번호가 잘못되었습니다. 다시 한 번 확인해주세요.",
+  },
+  valid: {
+    color: "black5",
+    inputState: "valid",
+  },
+});
```

Comment:
```suggestion
const verificationStateObject = (
  timeState: TimeState,
): VerificationStateObject => ({
  initial: {
    color: "black5",
    inputState: undefined,
    text: "인증번호가 전송되었습니다.",
  },
  invalid: {
    color: "danger1",
    inputState: "invalid",
    text:
      timeState === "end"
        ? "인증시간을 초과하였습다. 재인증을 시도해주세요."
        : "인증 번호가 잘못되었습니다. 다시 한 번 확인해주세요.",
  },
  valid: {
    color: "black5",
    inputState: "valid",
  },
} as const);
```
`as const` 키워드 활용하면 따로 타입적어주지 않아도 괜찮을것 같아요!

Reply:
`as`는 최대한 사용하고 싶지 않아서 직접 타입을 하나 만들어서 사용했습니다 ! 타입을 따로 선언하지 않고 `as const`를 사용했을때의 장점을 알 수 있을까요 ?

Reply:
해당값이 상수 값이고, `as const`를 사용해서 key 값의 타입과, 해당하는 value의 타입을 strict 하게 가져갈 수 있어요.
코드가 더 짧아진다는것도 장점인것 같아요.
특정 코드를 강제로 변환하는 type assertion 역할의 `as`는 지양하는게 맞다는 의견에 동의하지만, 값을 상수로 취급할때 사용하는 `as const`는 사용하는게 좋다고 생각해요!

Reply:
넵 코드가 짧아지는게 좋은거 같네요 ! 수정해서 올리겠습니다 !